### DISCUSSION TITLE
connecting USB while battery connected

### DISCUSSION THREAD

**Discussion by CH23-** [joeycastillo/Sensor-Watch#227] [Q&A] [answered]
Before I modify the watch module plastic to allow connecting the micro USB to the PCB, I would like to know if it can function and be flashed when the battery is still attached.

**Comment by joeycastillo** [comment, marked as answer]
**WARNING: You should absolutely not do this**, and there are (admittedly tiny) warnings on the PCB advising you against it. If you plug in USB while the battery is in the case, you will back-power the battery, which can lead to overheating or fire.

For background: when the board is clipped in with a coin cell, the battery is touching the power contacts and directly powering the board with 3 volts. When powered over USB, the board is getting power from the on-board 3.3 volt regulator, which means the power contacts are energized with 3.3 volts. The Sensor Watch board includes _no electrical isolation_ between these two systems for powering the watch, instead relying on the fact that the USB connector is mechanically inaccessible when the board is clipped in with a battery. 

This mechanical interlock is sufficient to ensure that the battery is never backpowered — unless you modify the watch module's plastic, which is why I am recommending that you not do that.

**Reply by CH23-** [reply]
Hi Joey, thank you for you incredibly fast answer, I failed to read the tiny warnings, sorry for that!

Would this work if only D- and D+ are connected (and a resistor cable-side between VCC/ID/GND so a USB host reads it as connected)?

**Reply by joeycastillo** [reply]
This would work! And in my experience (at least on a Mac) no resistor pull was required; D+, D- and ground were sufficient :)

**Reply by CH23-** [reply]
I cut the VCC line, but when I now connect it to my computer (battery inserted) it turns off completely, not responding to reset. I triple-checked, and there is no power coming through VCC from USB. also checked the battery, this reads 2.88v. Could it be that something on board switches to 5v power mode?

**Reply by joeycastillo** [reply]
Oh dang can you double check that the USB cable’s metal shield isn’t contacting the metal clippy thing on the back of the movement? That back plate is at 3V and the shield is tied to ground, which would short the battery and match what you’re describing.

**Reply by CH23-** [reply]
"surely it's not a grounding issue"

it was indeed a grounding issue. DREMELed away a bit more of the shield and presto, it's working!

Let me take this opportunity to thank you for this really fun project, although I'm not much of a developer, I love having such a fun (semi) smart watch on my wrist.
